src/index.ts

```diff
@@ -0,0 +1,103 @@
+import Environments from './structures/environments'
+
+type Nullable<T> = T | null
+
+interface IConversionFunction<T> {
+  (value: string): Nullable<T>
+}
+
+function _getAs<T> (fn: IConversionFunction<T>) {
+  return (names: string | string[], fallback: Nullable<string>) => {
+    const value = get(names, fallback)
+    if (!value) return value
```

Comment:
This is testing agains a `falsy` value but the current behaviour is testing only agains `null` values.
This change causes the following breaking change:

Current implementation:
```js
// process.env.FOOBAR = ''
env.get.int('FOOBAR', 0) // Number(0)
env.get.int('FOOBAR', '0') // Number(0)
env.get.int('FOOBAR', false) // NaN
env.get.int('FOOBAR', true) // NaN
```

Proposed implementation:
```js
// process.env.FOOBAR = ''
env.get.int('FOOBAR', 0) // Number(0)
env.get.int('FOOBAR', '0') // Number(0)
env.get.int('FOOBAR', false) // Boolean(false)
env.get.int('FOOBAR', true) // NaN
```

There's room for adding an unitary test for this you if may.

Reply:
Hum! Ok I Got it! Will correct here and add a test to it

Reply:
Great

Reply:
Fixed!